search/includes/classes/class-search.php

```diff
@@ -2401,6 +2417,67 @@ public function filter__ep_enable_do_weighting( $should_do_weighting, $weight_co
 		return false;
 	}
 
+	/**
+	 * This is used to toggle the option that stores the existence of custom search results.
+	 * Hooks onto {$new_status}_ep-pointer.
+	 *
+	 * @param int $post_id Post ID.
+	 * @param WP_Post $post Post object.
+	 * @param string $old_status Old post status.
+	 */
+	public function set_custom_results_existence_cache( $post_id, $post, $old_status = '' ) {
+		$custom_results_existence = $this->get_cached_custom_results_existence();
+		$option_cache_key         = 'vip_custom_results_existence';
+
+		if ( 'publish' === $post->post_status && '1' !== $custom_results_existence ) {
+			// Only set it to 1 if it's not already 1.
+			update_option( $option_cache_key, '1' );
+		} elseif ( 'trash' === $post->post_status && '1' === $custom_results_existence ) {
+			// If it's trashed, we need to query if there are any more CSRs left and if the option is 1.
+			$custom_results = $this->is_custom_results_exist();
+			if ( ! $custom_results ) {
+				update_option( $option_cache_key, '0' );
+			}
+		}
+	}
+
+	/**
+	 * Return the option that caches the existence of custom search results. If the option does not exist,
+	 * backfill it.
+	 *
+	 * @return string $custom_results_existence String boolean of whether custom search results exist.
+	 */
+	public function get_cached_custom_results_existence() {
+		$option_cache_key         = 'vip_custom_results_existence';
+		$custom_results_existence = get_option( $option_cache_key );
+
+		if ( false === $custom_results_existence ) {
+			$custom_results_existence = $this->is_custom_results_exist() ? '1' : '0';
+			update_option( $option_cache_key, $custom_results_existence );
+		}
+
+		return $custom_results_existence;
+	}
+
+	/**
+	 * Perform a WP_Query lookup to see if Custom Search Results are present.
+	 *
+	 * @return bool $exist If there are any Custom Search Results or not.
+	 */
+	public function is_custom_results_exist() {
+		$args           = [
+			'post_type'           => 'ep-pointer',
+			'post_status'         => 'publish',
+			'fields'              => 'ids',
+			'posts_per_page'      => 1,
+			'no_found_rows'       => true,
+			'ignore_sticky_posts' => true,
+		];
+		$custom_results = new \WP_Query( $args );
```

Comment:
@rinatkhaziev @rebeccahum a question - given the issue (https://github.com/Automattic/vip-go-mu-plugins/pull/3771) we observed today, would it make sense to add, above this line, something like 

```php
add_filter( 'ep_skip_query_integration', '__return_true', PHP_INT_MAX, 0 );
```
to ensure this query doesn't trigger somehow another Elasticpress call like we had in mentioned bug? I am not saying #3771 is not a proper solution, but maybe ensuring this `WP_Query` never goes with Elasticpress (what would cause this infinite loop).
After this line `remove_filter` call would be needed
```php
remove_filter( 'ep_skip_query_integration', '__return_true', PHP_INT_MAX );
```

(side note: probably it would make sense to use dedicated class method instead of `__return_true` in case any other piece of code would be doing the same using `__return_true` callback)